sdk/maps/azure-maps-weather/README.md

```diff
@@ -0,0 +1,232 @@
+# Azure Maps SDK Weather client library for Java
+
+Azure Maps SDK Weather client library for Java.
+
+This package contains the Azure Maps SDK Weather client library which contains Azure Maps Weather APIs. For documentation on how to use this package, please see [Azure Maps Weather SDK for Java](https://docs.microsoft.com/rest/api/maps/weather).
+
+[Source code][source] | [API reference documentation][docs] | [REST API documentation][rest_docs] | [Product documentation][product_docs] | [Samples][samples]
+
+## Documentation
+
+Various documentation is available to help you get started
+
+- [API reference documentation][docs]
+
+## Getting started
+
+### Prerequisites
+
+- [Java Development Kit (JDK)][jdk] with version 8 or above
+- [Azure Subscription][azure_subscription]
+
+### Adding the package to your product
+
+[//]: # ({x-version-update-start;com.azure:azure-maps-weather;current})
+```xml
+<dependency>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-maps-weather</artifactId>
+    <version>1.0.0-beta.1</version>
+</dependency>
+```
+[//]: # ({x-version-update-end})
+
+### Include the recommended packages
+
+Azure Maps Libraries require a `TokenCredential` implementation for authentication and an `HttpClient` implementation for HTTP client.
+
+[Azure Identity][azure_identity] package and [Azure Core Netty HTTP][azure_core_http_netty] package provide the default implementation.
+
+### Authentication
+
+By default, Azure Active Directory token authentication depends on correct configure of following environment variables.
+
+- `AZURE_CLIENT_ID` for Azure client ID.
+- `AZURE_TENANT_ID` for Azure tenant ID.
+- `AZURE_CLIENT_SECRET` or `AZURE_CLIENT_CERTIFICATE_PATH` for client secret or client certificate.
+
+In addition, Azure subscription ID can be configured via environment variable `AZURE_SUBSCRIPTION_ID`.
+
+With above configuration, `azure` client can be authenticated by following code:
+
+```java com.azure.maps.weather.sync.builder.ad.instantiation
+// Authenticates using Azure AD building a default credential
+// This will look for AZURE_CLIENT_ID, AZURE_TENANT_ID, and AZURE_CLIENT_SECRET env variables
+DefaultAzureCredential tokenCredential = new DefaultAzureCredentialBuilder().build();
+
+// Creates a client
+WeatherClient client = new WeatherClientBuilder()
+    .credential(tokenCredential)
+    .weatherClientId(System.getenv("MAPS_CLIENT_ID"))
```

Comment:
nit
```suggestion
    .weatherClientId(System.getenv("WEATHER_CLIENT_ID"))
```

Reply:
In the setup for Azure Maps, the environment variable is maps_client_id across all modules including search, route render etc!